install.sh

```diff
@@ -608,49 +619,75 @@ main() {
   PREFIX=${PREFIX:-/usr/local}
   CACHE_DIR=$(cache_dir)
   mkdir -p "$CACHE_DIR"
+  METHOD=${METHOD:-detect}
   INSTALL_DIR=$PREFIX/lib/d2
 
+  case $METHOD in
+    detect)
+      case "$OS" in
+        macos)
```

Comment:
is OS the only criteria for triggering homebrew install? is there a check that `brew` is in their path?